function_calling/src/tools.py

```diff
@@ -0,0 +1,163 @@
+import os
```

Comment:
I would define here all the classes and functions of the notebook that haven't been defined.